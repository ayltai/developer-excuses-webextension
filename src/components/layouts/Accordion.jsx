import { makeStyles, ExpansionPanel, ExpansionPanelDetails, ExpansionPanelSummary, Typography, } from '@material-ui/core';
import { ExpandMore, } from '@material-ui/icons';
import PropTypes from 'prop-types';
import React from 'react';

export const Accordion = props => {
    const classes = makeStyles(() => ({
        block : {
            display : 'block',
        },
    }))();
    return (
        <ExpansionPanel square>
            <ExpansionPanelSummary expandIcon={<ExpandMore />}>
                <Typography variant='h6'>{props.label}</Typography>
            </ExpansionPanelSummary>
            <ExpansionPanelDetails className={classes.block}>
                {props.children}
            </ExpansionPanelDetails>
        </ExpansionPanel>
    );
};

Accordion.propTypes = {
    children : PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.arrayOf(PropTypes.node),
    ]),
    label    : PropTypes.string,
};
