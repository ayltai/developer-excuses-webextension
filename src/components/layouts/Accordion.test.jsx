import { mount, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { Accordion, } from './Accordion';

describe('<Accordion />', () => {
    it('renders correctly', () => {
        expect(renderer.create(<Accordion label='dummy' />).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <Accordion label='dummy'>
                <div />
            </Accordion>
        );
    });
});
