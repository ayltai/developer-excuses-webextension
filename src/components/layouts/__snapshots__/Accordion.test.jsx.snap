// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders correctly 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root MuiPaper-elevation1"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="MuiButtonBase-root MuiExpansionPanelSummary-root"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiExpansionPanelSummary-content"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        dummy
      </h6>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiExpansionPanelDetails-root makeStyles-block-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
